echopype/echodata/zarr_combine.py

```diff
@@ -916,7 +917,9 @@ def _modify_prov_filenames(
 
         # obtain the filenames zarr array
         zarr_filenames = zarr.open_array(
-            zarr_path + "/Provenance/filenames", mode="r+", storage_options=storage_options
+            os.path.join(zarr_path, "Provenance", "filenames"),
```

Comment:
This is a good idea, however I'm concerned that if the `zarr_path` is a url base to a bucket, there will be a problem on windows machine. `s3://my-bucket/data.zarr` can become `s3://my-bucket/data.zarr\Provenance\filenames`. I feel like there should be a function to handle this within your class.

Reply:
You are correct, I just tried it out on a Window's machine and the output is not correct. It will output the following: `s3://my-bucket/data.zarr\\Provenance\\filenames`. Do you have a function that already handles this? If not, I guess I can just create a function that determines if `s3://` is at the beginning of `zarr_path` and if it is concatenate appropriately. Do you know a better way to do this?

Reply:
The problem with my solution is that it limits us to `s3`. I am not sure what the other paths look like for other cloud storage options.

Reply:
Anything with storage stuff will always have `://` so you can check for that in the path string and adjust as appropriate for the path concatenation. Currently we don't have a function that handles this. `sanitize_file_path` is the closest one: https://github.com/OSOceanAcoustics/echopype/blob/b58d1eb58dcc6dc33e26d196d809ae55c2ad3155/echopype/utils/io.py#L120. You can do similar things.

Reply:
I suppose we could just check for the main cloud storage options: AWS is `"s3://"`, Azure is `"az://"`, and Google Cloud storage is `"gs://"` (at least from what I can find).

Reply:
Ah, for some reason your comment did not show up! I can go ahead and create my own function using the logic `if "://" in file_path:`. Thank you for the suggestion.